backend/backend.c

```diff
@@ -203,19 +203,22 @@ struct wlr_backend *wlr_backend_autocreate(struct wl_display *display,
 				wlr_log(L_ERROR, "failed to start backend '%s'", name);
 				wlr_backend_destroy(backend);
 				wlr_session_destroy(session);
+				free(names);
 				return NULL;
 			}
 
 			if (!wlr_multi_backend_add(backend, subbackend)) {
 				wlr_log(L_ERROR, "failed to add backend '%s'", name);
 				wlr_backend_destroy(backend);
 				wlr_session_destroy(session);
+				free(names);
 				return NULL;
 			}
 
 			name = strtok_r(NULL, ",", &saveptr);
 		}
 
+		free(names);
```

Comment:
(last push added this line, I hadn't looked deeply enough into what `names` was used for and it came up on recheck. Am done rechecking now)